frontend/ui/widget/statuswidget.lua

```diff
@@ -0,0 +1,653 @@
+        self.summary.rating = num
+        self:saveSummary()
+
+        for i = 1, num do
+            table.insert(stars_group, self.star:new { icon = "resources/icons/stats.star.full.png", callback = function() self:setStar(i) end })
+        end
+    else
+        num = 0
+    end
+
+    for i = num + 1, 5 do
+        table.insert(stars_group, self.star:new { callback = function() self:setStar(i) end })
+    end
+
+    table.insert(self.stars_container, stars_group)
+
+    UIManager:setDirty(nil, "partial")
+    return true
+end
+
+--TODO generate from table
+function StatusWidget:generateStatisticsGroup(width, height, days, average, pages)
+    local statistics_container = CenterContainer:new {
+        dimen = Geom:new { w = width, h = height },
+    }
+
+    local statistics_group = VerticalGroup:new { align = "left" }
+    local titles_group = HorizontalGroup:new { align = "center" }
+    local data_group = HorizontalGroup:new { align = "center" }
+
+    local tile_width = width * 0.33333
+    local tile_height = height * 0.5
+
+    local title_days_container = CenterContainer:new {
+        dimen = Geom:new { w = tile_width, h = tile_height },
+        TextWidget:new {
+            text = _("Days"),
+            face = self.small_font_face,
+        },
+    }
+    local title_time_container = CenterContainer:new {
+        dimen = Geom:new { w = tile_width, h = tile_height },
+        TextWidget:new {
+            text = _("Time"),
+            face = self.small_font_face,
+        },
+    }
+    local title_read_pages_container = CenterContainer:new {
+        dimen = Geom:new { w = tile_width, h = tile_height },
+        TextWidget:new {
+            text = _("Read pages"),
+            face = self.small_font_face,
+        }
+    }
+
+    table.insert(titles_group, title_days_container)
+    table.insert(titles_group, title_time_container)
+    table.insert(titles_group, title_read_pages_container)
+
+
+    local days_container = CenterContainer:new {
+        dimen = Geom:new { w = tile_width, h = tile_height },
+        TextWidget:new {
+            text = days,
+            face = self.medium_font_face,
+        },
+    }
+    local average_time_container = CenterContainer:new {
+        dimen = Geom:new { w = tile_width, h = tile_height },
+        TextWidget:new {
+            text = average,
+            face = self.medium_font_face,
+        },
+    }
+    local read_pages_container = CenterContainer:new {
+        dimen = Geom:new { w = tile_width, h = tile_height },
+        TextWidget:new {
+            text = pages,
+            face = self.medium_font_face,
+        }
+    }
+
+    table.insert(data_group, days_container)
+    table.insert(data_group, average_time_container)
+    table.insert(data_group, read_pages_container)
+
+    table.insert(statistics_group, titles_group)
+    table.insert(statistics_group, data_group)
+
+    table.insert(statistics_container, statistics_group)
+    return statistics_container
```

Comment:
How about rewrite the return like the following? I personally think it's more readable:

```
return CenterContainer:new{
	dimen = Geom:new{ w = width, h = height },
	VerticalGroup:new{
		align = "left"
		titles_group,
		data_group
	}
}
```